packages/algob/src/internal/util/deployer.ts

```diff
@@ -0,0 +1,69 @@
+import { mkAccountIndex } from "../../lib/account";
+import { AlgoOperator } from "../../lib/algo-operator";
+import { loadASAFile } from "../../lib/asa";
+import { loadCheckpointsRecursive } from "../../lib/script-checkpoints";
+import type {
+  Accounts,
+  AlgobDeployer,
+  AlgobRuntimeEnv,
+  ASADefs,
+  CheckpointRepo
+} from "../../types";
+import {
+  DeployerDeployMode,
+  DeployerRunMode
+} from "../deployer";
+import { txWriter, TxWriterImpl } from "../tx-log-writer";
+
+export interface DeployerConfig {
+  runtimeEnv: AlgobRuntimeEnv
+  cpData: CheckpointRepo
+  asaDefs: ASADefs
+  algoOp: AlgoOperator
+  accounts: Accounts
+  txWriter: txWriter
+}
+
+export function mkDeployer (
+  allowWrite: boolean,
+  deployerConfig: DeployerConfig
+): AlgobDeployer {
+  if (allowWrite) {
+    return new DeployerDeployMode(
+      deployerConfig.runtimeEnv,
+      deployerConfig.cpData,
+      deployerConfig.asaDefs,
+      deployerConfig.algoOp,
+      deployerConfig.accounts,
+      deployerConfig.txWriter);
+  }
+  return new DeployerRunMode(
+    deployerConfig.runtimeEnv,
+    deployerConfig.cpData,
+    deployerConfig.asaDefs,
+    deployerConfig.algoOp,
+    deployerConfig.accounts,
+    deployerConfig.txWriter);
+}
+
+// intialize deployer config obj
+export class MkDeployerConfig implements DeployerConfig {
+  runtimeEnv: AlgobRuntimeEnv;
+  cpData: CheckpointRepo;
+  asaDefs: ASADefs;
+  algoOp: AlgoOperator;
+  txWriter: txWriter;
+  accounts: Accounts;
+
+  constructor (
+    runtimeEnv: AlgobRuntimeEnv,
+    algoOp: AlgoOperator
+  ) {
```

Comment:
this can be put in a single line as there are only two variables. `constructor (runtimeEnv: AlgobRuntimeEnv, algoOp: AlgoOperator) {`

Reply:
Let's see if we pass entire object to deployer. If yes then we'll need to pass `asaDefs` (maybe, not sure) as well. Will update accordingly.

Reply:
Updated.